src/helpers/__tests__/utils.test.ts

```diff
@@ -76,20 +77,66 @@ describe('utils', () => {
+    describe('proxy configuration', () => {
+      test('should have a ProxyAgent by default', async () => {
+        jest.spyOn(axios, 'create').mockImplementation((() => (args: AxiosRequestConfig) => args.httpsAgent) as any)
+        const requestOptions = {
+          apiKey: 'apiKey',
+          appKey: 'applicationKey',
+          baseUrl: 'http://fake-base.url/',
+        }
+        const request = ciUtils.getRequestBuilder(requestOptions)
+        const fakeEndpoint = fakeEndpointBuilder(request)
+        const httpsAgent = await fakeEndpoint()
+        expect(httpsAgent).toBeDefined()
+        expect(httpsAgent).toBeInstanceOf(ProxyAgent)
+      })
+      test('should not have a proxy agent if proxy env vars support is disabled', async () => {
+        jest.spyOn(axios, 'create').mockImplementation((() => (args: AxiosRequestConfig) => args.httpsAgent) as any)
+        const requestOptions = {
+          apiKey: 'apiKey',
+          appKey: 'applicationKey',
+          baseUrl: 'http://fake-base.url/',
+          disableEnvironmentVariables: true,
+        }
+        const request = ciUtils.getRequestBuilder(requestOptions)
+        const fakeEndpoint = fakeEndpointBuilder(request)
+        const httpsAgent = await fakeEndpoint()
+        expect(httpsAgent).toBeUndefined()
+      })
+
+      test('should add proxy configuration when explicitly defined', async () => {
+        jest.spyOn(axios, 'create').mockImplementation((() => (args: AxiosRequestConfig) => args.httpsAgent) as any)
+        const proxyOpts: ciUtils.ProxyConfiguration = {protocol: 'http', host: '1.2.3.4', port: 1234}
+        const requestOptions = {
+          apiKey: 'apiKey',
+          appKey: 'applicationKey',
+          baseUrl: 'http://fake-base.url/',
+          proxyOpts,
+        }
+        const request = ciUtils.getRequestBuilder(requestOptions)
+        const fakeEndpoint = fakeEndpointBuilder(request)
+        const httpsAgent = await fakeEndpoint()
+        expect(httpsAgent).toBeDefined()
+        expect((httpsAgent as any).proxyUri).toBe('http://1.2.3.4:1234')
+      })
+
+      test('should add proxy configuration when explicitly defined even without env vars', async () => {
+        jest.spyOn(axios, 'create').mockImplementation((() => (args: AxiosRequestConfig) => args.httpsAgent) as any)
+        const proxyOpts: ciUtils.ProxyConfiguration = {protocol: 'http', host: '1.2.3.4', port: 1234}
+        const requestOptions = {
+          apiKey: 'apiKey',
+          appKey: 'applicationKey',
+          baseUrl: 'http://fake-base.url/',
+          disableEnvironmentVariables: true,
+          proxyOpts,
+        }
+        const request = ciUtils.getRequestBuilder(requestOptions)
+        const fakeEndpoint = fakeEndpointBuilder(request)
+        const httpsAgent = await fakeEndpoint()
+        expect(httpsAgent).toBeDefined()
+        expect((httpsAgent as any).proxyUri).toBe('http://1.2.3.4:1234')
+      })
```

Comment:
Yes the problem is not only to modify the `process.env` var but also to verify that the proxy is correctly called. Using inspiration from `proxy-agent` test suite, I can come up with a test using an https.Server and a proxy defined with the npmjs [proxy package](https://www.npmjs.com/package/proxy), and add spys on the callback functions to verify the behavior.

It's some work but I think it has some value to avoid regression; I'll keep that for a later PR though.

Reply:
:+1: Thanks for the explanations